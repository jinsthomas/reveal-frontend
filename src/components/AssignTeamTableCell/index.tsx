import reducerRegistry from '@onaio/redux-reducer-registry';
import { keys, values } from 'lodash';
import React, { MouseEvent, useState } from 'react';
import { connect } from 'react-redux';
import { Button, Popover, PopoverBody, PopoverHeader } from 'reactstrap';
import { Store } from 'redux';
import { stopPropagationAndPreventDefault } from '../../helpers/utils';
import organizationsReducer, {
  getOrganizationsById,
  Organization,
  reducerName as organizationsReducerName,
} from '../../store/ducks/opensrp/organizations';
import Loading from '../page/Loading';
reducerRegistry.register(organizationsReducerName, organizationsReducer);

/** Interface for Assign Teams cell props */
export interface AssignTeamCellProps {
  jurisdictionId: string;
  organizationsById: { [key: string]: Organization } | null;
  planId: string;
}

/** Component that will be rendered in IRS planning table cells
 * showing the button and card in the Assign Teams column
 */
const AssignTeamTableCell = (props: AssignTeamCellProps) => {
  const { jurisdictionId, organizationsById } = props;
  const [isActive, setIsActive] = useState<boolean>(false);

  const onPlanAssignmentButtonClick = (e: MouseEvent) => {
    stopPropagationAndPreventDefault(e);
    setIsActive(!isActive);
  };

  const AssignTeamButton = (
    <Button
      color="primary"
      id={getButtonId(jurisdictionId)}
      onClick={onPlanAssignmentButtonClick}
      size="sm"
    >
      Assign Teams
    </Button>
  );

  const organizationsArray: Organization[] | null =
    organizationsById &&
    keys(organizationsById)
      .map((o: string) => organizationsById[o])
      .filter(o => !!o);

  const AssignPopover = (
    <Popover target={getButtonId(jurisdictionId)} isOpen={isActive}>
      <PopoverHeader>Select Teams to Assign</PopoverHeader>
      <PopoverBody>
        <h1>Here be Teams!</h1>
        {organizationsArray ? (
          <ul>
            {organizationsArray.map((organization: Organization, i: number) => (
              <li key={i}>{organization.name}</li>
            ))}
          </ul>
        ) : (
          <Loading />
        )}
      </PopoverBody>
    </Popover>
  );

  return (
    <div>
      {AssignTeamButton}
      {AssignPopover}
    </div>
  );
};

const getButtonId = (jurisdictionId: string): string => `plan-assignment-${jurisdictionId}`;
export { AssignTeamTableCell };

/** map state to props
 * @param {partial<store>} - the redux store
 * @param {AssignTeamCellProps} ownProps - the props
 *
 * @returns {AssignTeamCellProps} - ownProps and organizationsById
 */
const mapStateToProps = (state: Partial<Store>, ownProps: any): AssignTeamCellProps => {
  const organizationsById = getOrganizationsById(state);
  return {
    ...ownProps,
    organizationsById,
  } as AssignTeamCellProps;
};

/** map props to actions that may be dispatched by component */
const mapDispatchToProps = {};

/** Create Connected AssignTeamTableCell */
const ConnectedAssignTeamTableCell = connect(
  mapStateToProps,
  mapDispatchToProps
)(AssignTeamTableCell);

export default ConnectedAssignTeamTableCell;
