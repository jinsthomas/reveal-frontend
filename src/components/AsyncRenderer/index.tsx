import React, { PropsWithChildren } from 'react';
import { AsyncState, IfFulfilled, IfPending, PromiseFn, useAsync } from 'react-async';
import { PLEASE_PROVIDE_CUSTOM_COMPONENT_TO_RENDER } from '../../constants';
import Loading from '../page/Loading';

/** props for AsyncRender component */
export interface AsyncRendererProps<TData, TPromiseFunctionProps> {
  data: TData[] /** interface describing data to be consumed by render component */;
  promiseFn: PromiseFn<TData[]> /** react-async-type asynchronous function */;
  promiseFnProps: TPromiseFunctionProps /** options to be passed to promiseFn */;
  ifLoadingRender: (
    state: AsyncState<TData[]>
  ) => React.ReactNode /** renderProp invoked while promise returned by promiseFn is pending */;
  ifFulfilledRender: (
    state: AsyncState<TData[]>
  ) => React.ReactNode /** renderProp invoked after promise returned by promiseFn is fulfilled */;
}

// tslint:disable-next-line: no-empty-destructuring
const defaultPromiseFn = async ({}) => [];

/** default props for AsyncRenderer */
const defaultAsyncRenderProps: AsyncRendererProps<any, any> = {
  data: [],
  ifFulfilledRender: () => <div>{PLEASE_PROVIDE_CUSTOM_COMPONENT_TO_RENDER}</div>,
  ifLoadingRender: () => <Loading />,
  promiseFn: defaultPromiseFn,
  promiseFnProps: {},
};

/** helps Dry out trivial react-async rendering */
export const AsyncRenderer = <TData, TPromiseFunctionProps>({
  data,
  ifFulfilledRender,
  ifLoadingRender,
  promiseFn,
  promiseFnProps,
}: PropsWithChildren<
  AsyncRendererProps<TData, TPromiseFunctionProps>
> = defaultAsyncRenderProps) => {
  const loadPractitionersState = useAsync<TData[]>(promiseFn, promiseFnProps);

  React.useEffect(() => {
    if (data.length > 0) {
      loadPractitionersState.setData(data);
    }
  }, []);

  return (
    <>
      <IfPending state={loadPractitionersState}>
        {ifLoadingRender(loadPractitionersState)}
      </IfPending>
      <IfFulfilled state={loadPractitionersState}>
        {ifFulfilledRender(loadPractitionersState)}
      </IfFulfilled>
    </>
  );
};
