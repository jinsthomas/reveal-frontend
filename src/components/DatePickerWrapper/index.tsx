import { FieldProps } from 'formik';
import React from 'react';
import DatePicker, { ReactDatePickerProps } from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';

const DatePickerWrapper = (props: ReactDatePickerProps & FieldProps) => {
  const handleChange = (date: any) => {
    props.form.setFieldValue(props.field.name, date);
  };

  return <DatePicker selected={props.field.value} onChange={handleChange} isClearable={true} />;
};

export default DatePickerWrapper;
