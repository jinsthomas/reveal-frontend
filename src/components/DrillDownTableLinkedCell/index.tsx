import { DropDownCellProps } from '@onaio/drill-down-table';
import React from 'react';
import { Link } from 'react-router-dom';
import { FOCUS_INVESTIGATION_URL } from '../../constants';

/** Component that will be rendered in drop-down table cells showing a link
 * that moves you to the next hierarchical level.
 */
const DrillDownTableLinkedCell: React.ElementType<DropDownCellProps> = (
  props: DropDownCellProps
) => {
  const { cellValue, hasChildren } = props;
  return (
    <div>
      {hasChildren ? (
        <Link to={`${FOCUS_INVESTIGATION_URL}/${cellValue}`}>{cellValue}</Link>
      ) : (
        cellValue
      )}
    </div>
  );
};

export default DrillDownTableLinkedCell;
