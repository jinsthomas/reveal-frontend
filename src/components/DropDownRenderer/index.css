.filter-bar-btns {
  color: black;
  min-height: 35px;
}

.filter-bar-btns span {
  font-size: 16px !important;
  margin: 5px;
}
