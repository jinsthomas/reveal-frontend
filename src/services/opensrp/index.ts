import { OpenSRPService as OpenSRPServiceWeb } from '@opensrp/server-service';
import { IncomingHttpHeaders } from 'http';
import { OPENSRP_API_BASE_URL } from '../../configs/env';
import store from '../../store';
import { getAccessToken } from '../../store/selectors';

/** allowed http methods */
type HTTPMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';

/** get default HTTP headers for OpenSRP service */
export function getDefaultHeaders(
  accept: string = 'application/json',
  authorizationType: string = 'Bearer',
  contentType: string = 'application/json;charset=UTF-8'
): IncomingHttpHeaders {
  return {
    accept,
    authorization: `${authorizationType} ${getAccessToken(store.getState())}`,
    'content-type': contentType,
  };
}

/** interface to describe URL params object */
export interface URLParams {
  [key: string]: string | number | boolean;
}

/** converts filter params object to string
 * @param {URLParams} obj - the object representing filter params
 * @returns {string} filter params as a string
 */
export function getFilterParams(obj: URLParams | {}): string {
  return Object.entries(obj)
    .map(([key, val]) => `${key}:${val}`)
    .join(',');
}

/** get payload for fetch
 * @param {AbortSignal} signal - signal object that allows you to communicate with a DOM request
 * @param {HTTPMethod} method - the HTTP method
 * @returns the payload
 */

export function getPayloadOptions(_: AbortSignal, method: HTTPMethod) {
  return {
    headers: getDefaultHeaders() as HeadersInit,
    method,
  };
}

/** The OpenSRP service class
 * Extends class OpenSRPService from OpenSRPService web
 * Sample usage:
 * -------------
 * const service = new OpenSRPService('the-endpoint');
 *
 * **To list all objects**: service.list()
 *
 * **To get one object**: service.read('the-object-identifier')
 *
 * **To create a new object**: service.create(theObject)
 *
 * **To update an object**: service.update(theObject)
 */
export class OpenSRPService extends OpenSRPServiceWeb {
  constructor(
    endpoint: string,
    baseURL: string = OPENSRP_API_BASE_URL,
    getPayload: typeof getPayloadOptions = getPayloadOptions
  ) {
    super(baseURL, endpoint, getPayload);
  }
  /** create method
   * Send a POST request to the general endpoint containing the new object data
   * Successful requests will result in a HTTP status 201 response with no body
   * @param {T} data - the data to be POSTed
   * @param {params} params - the url params object
   * @param {HTTPMethod} method - the HTTP method
   * @returns the object returned by API
   */
  public async create<T>(data: T, params: any = null, method: HTTPMethod = 'POST'): Promise<{}> {
    const url = OpenSRPService.getURL(this.generalURL, params);
    const payload = {
      ...this.getOptions(this.signal, method),
      'Cache-Control': 'no-cache',
      Pragma: 'no-cache',
      body: JSON.stringify(data),
    };
    const response = await fetch(url, payload);

    if (!response.ok || response.status !== 201) {
      throw new Error(
        `OpenSRPService create on ${this.endpoint} failed, HTTP status ${response.status}`
      );
    }

    return {};
  }
}
