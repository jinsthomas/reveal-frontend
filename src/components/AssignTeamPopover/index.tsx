import { keys } from 'lodash';
import React, { MouseEvent } from 'react';
import { Button, Form, Popover, PopoverBody, PopoverHeader } from 'reactstrap';
import OrganizationSelect from '../../containers/forms/OrganizationSelect';
import { stopPropagationAndPreventDefault } from '../../helpers/utils';
import { Organization } from '../../store/ducks/opensrp/organizations';

/** Interface for AssignTeamPopover */
export interface AssignTeamPopoverProps {
  formName: string;
  isActive: boolean;
  jurisdictionId: string;
  onClearAssignmentsButtonClick: (e: MouseEvent) => void;
  onSaveAssignmentsButtonClick: (e: MouseEvent) => void;
  onToggle: () => void;
  organizationsById: { [key: string]: Organization } | null;
  planId: string;
  target: string;
}

const AssignTeamPopover = (props: AssignTeamPopoverProps) => {
  const {
    formName,
    isActive,
    jurisdictionId,
    onClearAssignmentsButtonClick,
    onSaveAssignmentsButtonClick,
    onToggle,
    organizationsById,
    planId,
    target,
  } = props;

  const organizationsArray: Organization[] | null =
    organizationsById &&
    keys(organizationsById)
      .map((o: string) => organizationsById[o])
      .filter(o => !!o);

  const organizationSelectProps = {
    jurisdictionId,
    name: formName,
    planId,
  };

  const popoverProps = {
    isOpen: isActive,
    onClick: (e: React.MouseEvent) => stopPropagationAndPreventDefault(e),
    target,
    toggle: () => onToggle(),
  };

  return (
    <Popover {...popoverProps}>
      <PopoverHeader>Select Teams to Assign</PopoverHeader>
      <PopoverBody>
        {organizationsArray ? (
          <Form name={formName}>
            <OrganizationSelect {...organizationSelectProps} />
            <Button color="default" onClick={onClearAssignmentsButtonClick} size="xs">
              Clear
            </Button>
            <Button color="primary" onClick={onSaveAssignmentsButtonClick} size="xs">
              Save
            </Button>
          </Form>
        ) : (
          <p>No teams loaded...</p>
        )}
      </PopoverBody>
    </Popover>
  );
};

export default AssignTeamPopover;
