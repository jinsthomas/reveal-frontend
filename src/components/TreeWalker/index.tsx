import React, { Fragment, useEffect, useState } from 'react';
import { displayError } from '../../helpers/errors';
import { getFilterParams, OpenSRPService, URLParams } from '../../services/opensrp';

/** interface for jurisdiction options
 * These are received from the OpenSRP API
 */
interface OpenSRPJurisdiction {
  id: string;
  properties: {
    status: string;
    name: string;
    geographicLevel: number;
    parentId?: string;
    version: string | number;
  };
  serverVersion: number;
  type: 'Feature';
}

interface SimpleProps {
  apiEndpoint: string;
  jurisdictionId: string;
  params: URLParams;
  serviceClass: typeof OpenSRPService;
}

const defaultParams = {
  is_jurisdiction: true,
  return_geometry: false,
};

const defaultSimpleProps: SimpleProps = {
  apiEndpoint: 'location/findByProperties',
  jurisdictionId: '',
  params: defaultParams,
  serviceClass: OpenSRPService,
};

const getAncestors = async (
  jurisdiction: OpenSRPJurisdiction,
  path: OpenSRPJurisdiction[] = []
): Promise<OpenSRPJurisdiction[] | null> => {
  if (!path.includes(jurisdiction)) {
    path.unshift(jurisdiction);
  }

  if (jurisdiction.properties.geographicLevel === 0 || !jurisdiction.properties.parentId) {
    return path;
  }

  const service = new OpenSRPService('location');
  const parentJurisdiction = await service
    .read(jurisdiction.properties.parentId, defaultParams)
    .then((response: OpenSRPJurisdiction) => {
      if (response) {
        return response;
      }
    })
    .catch((error: Error) => displayError(error));

  if (!parentJurisdiction) {
    return null;
  } else {
    return getAncestors(parentJurisdiction, path);
  }
};

const Simple = (props: SimpleProps) => {
  const [current, setCurrent] = useState<OpenSRPJurisdiction[]>([]);
  const [selectedJurisdiction, setSelectedJurisdiction] = useState<OpenSRPJurisdiction | null>(
    null
  );
  const [hierarchy, setHierarchy] = useState<OpenSRPJurisdiction[]>([]);
  const [loading, setLoading] = useState<boolean>(true);
  const { apiEndpoint, jurisdictionId, params, serviceClass } = props;

  const service = new serviceClass(apiEndpoint);

  const parentId = selectedJurisdiction ? selectedJurisdiction.id : jurisdictionId;

  const propertiesToFilter = {
    status: 'Active',
    ...(parentId === '' && { geographicLevel: 0 }),
    ...(parentId !== '' && { parentId }),
  };

  const paramsToUse = {
    ...params,
    ...(Object.keys(propertiesToFilter).length > 0 && {
      properties_filter: getFilterParams(propertiesToFilter),
    }),
  };

  useEffect(() => {
    if (!selectedJurisdiction && jurisdictionId !== '') {
      const singleService = new OpenSRPService('location');
      singleService
        .read(jurisdictionId, params)
        .then((response: OpenSRPJurisdiction) => {
          if (response) {
            setSelectedJurisdiction(response);
            getAncestors(response)
              .then((path: OpenSRPJurisdiction[] | null) => {
                if (path) {
                  setHierarchy(path);
                }
              })
              .catch((error: Error) => displayError(error));
          }
        })
        .catch((error: Error) => displayError(error));
    }
  }, []);

  useEffect(() => {
    service
      .list(paramsToUse)
      .then((response: OpenSRPJurisdiction[]) => {
        if (response) {
          setCurrent(response);
        }
      })
      .finally(() => setLoading(false))
      .catch((error: Error) => displayError(error));
  }, [parentId]);

  if (loading === true) {
    return <Fragment>Loading...</Fragment>;
  }

  const loadMore = (item: OpenSRPJurisdiction, _: Event | React.MouseEvent) => {
    if (!hierarchy.includes(item)) {
      hierarchy.push(item);
    } else {
      // remove all elements in the array that come after item
      // hierarchy should include elements only up to the current item
      hierarchy.length = hierarchy.indexOf(item) + 1;
    }
    setHierarchy(hierarchy);
    setSelectedJurisdiction(item);
  };

  if (current.length > 0 || selectedJurisdiction !== null) {
    return (
      <Fragment>
        {selectedJurisdiction && (
          <Fragment>
            <h4>Currently Selected</h4>
            <span>{selectedJurisdiction.properties.name}</span>
          </Fragment>
        )}
        <h4>Path</h4>
        {hierarchy.map((item: OpenSRPJurisdiction, index: number) => (
          <span key={`jzz-${index}`} onClick={e => loadMore(item, e)}>
            {' '}
            {item.properties.name}
            {'>> '}
          </span>
        ))}
        <h4>Current children</h4>
        <ul>
          {current.map((item: OpenSRPJurisdiction, index: number) => (
            <li key={`jxx-${index}`} onClick={e => loadMore(item, e)}>
              {item.properties.name}
            </li>
          ))}
        </ul>
        ;
      </Fragment>
    );
  } else {
    return <Fragment>No Options</Fragment>;
  }
};

Simple.defaultProps = defaultSimpleProps;

export { Simple };
