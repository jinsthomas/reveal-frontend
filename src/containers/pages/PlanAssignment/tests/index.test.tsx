import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount } from 'enzyme';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Loading from '../../../../components/page/Loading';
import { defaultWalkerProps } from '../../../../components/TreeWalker';
import { limitTree, raKashikishiHAHC } from '../../../../components/TreeWalker/tests/fixtures';
import { OpenSRPService } from '../../../../services/opensrp';
import assignmentReducer, {
  fetchAssignments,
  reducerName as assignmentReducerName,
} from '../../../../store/ducks/opensrp/assignments';
import organizationsReducer, {
  fetchOrganizations,
  reducerName as organizationsReducerName,
} from '../../../../store/ducks/opensrp/organizations';
import planDefinitionReducer, {
  addPlanDefinition,
  reducerName as planDefinitionReducerName,
} from '../../../../store/ducks/opensrp/PlanDefinition';
import { plans } from '../../../../store/ducks/opensrp/PlanDefinition/tests/fixtures';
import {
  organization1,
  organization2,
  organization3,
} from '../../../../store/ducks/tests/fixtures';
import { assignments } from '../helpers/JurisdictionAssignmentForm/tests/fixtures';
import { PlanAssignment } from '../index';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

// register reducers
reducerRegistry.register(assignmentReducerName, assignmentReducer);
reducerRegistry.register(organizationsReducerName, organizationsReducer);
reducerRegistry.register(planDefinitionReducerName, planDefinitionReducer);

const history = createBrowserHistory();

describe('PlanAssignment', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    fetchMock.mockClear();
    fetchMock.resetMocks();
  });

  it('works as expected', async () => {
    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => limitTree);

    fetch.mockResponses(
      [JSON.stringify(assignments), { status: 200 }],
      [JSON.stringify([organization1, organization2, organization3]), { status: 200 }],
      [JSON.stringify([plans[0]]), { status: 200 }]
    );

    const callBack: any = jest.fn();
    const orgs = [organization1, organization2, organization3];
    const props = {
      ...defaultWalkerProps,
      addPlanActionCreator: addPlanDefinition,
      assignments,
      fetchAssignmentsActionCreator: fetchAssignments,
      fetchOrganizationsActionCreator: fetchOrganizations,
      history,
      location: {
        hash: '',
        pathname: `/assign/${plans[0].identifier}/${raKashikishiHAHC.id}`,
        search: '',
        state: {},
      },
      match: {
        isExact: true,
        params: { jurisdictionId: raKashikishiHAHC.id, planId: plans[0].identifier },
        path: '/assign/:planId/:jurisdictionId',
        url: `/assign/${plans[0].identifier}/${raKashikishiHAHC.id}`,
      },
      organizations: orgs,
      plan: plans[0],
      submitCallBackFunc: callBack,
      supersetService: supersetServiceMock,
    };

    const wrapper = mount(
      <MemoryRouter
        initialEntries={[
          {
            pathname: `/assign/${plans[0].identifier}/${raKashikishiHAHC.id}`,
            search: '',
            state: {},
          },
        ]}
      >
        <PlanAssignment {...props} />
      </MemoryRouter>
    );

    await flushPromises();
    wrapper.update();
    expect(fetch.mock.calls[0][0]).toEqual(
      'https://reveal-stage.smartregister.org/opensrp/rest/organization/assignedLocationsAndPlans?plan=356b6b84-fc36-4389-a44a-2b038ed2f38d'
    );
    expect(fetch.mock.calls[1][0]).toEqual(
      'https://reveal-stage.smartregister.org/opensrp/rest/organization'
    );
    expect(fetch.mock.calls[2][0]).toEqual(
      'https://reveal-stage.smartregister.org/opensrp/rest/plans/356b6b84-fc36-4389-a44a-2b038ed2f38d'
    );

    // expect(supersetServiceMock.mock.calls).toEqual([
    //   [
    //     1 /** this comes from the envs mock */,
    //     {
    //       adhoc_filters: [
    //         {
    //           clause: 'WHERE',
    //           comparator: '450fc15b-5bd2-468a-927a-49cb10d3bcac',
    //           expressionType: 'SIMPLE',
    //           operator: '==',
    //           subject: 'jurisdiction_id',
    //         },
    //       ],
    //       row_limit: 1,
    //     },
    //   ],
    // ]);

    expect(wrapper.find('TreeWalker').props()).toEqual({
      LoadingIndicator: Loading,
      OpenSRPServiceClass: OpenSRPService,
      addPlanActionCreator: addPlanDefinition,
      assignments,
      currentChildren: [],
      currentNode: null,
      fetchAssignmentsActionCreator: fetchAssignments,
      fetchOrganizationsActionCreator: fetchOrganizations,
      getAncestorsFunc: expect.any(Function),
      getChildrenFunc: expect.any(Function),
      hierarchy: [],
      history: expect.any(Object),
      jurisdictionId: '8d44d54e-8b4c-465c-9e93-364a25739a6d',
      labels: {
        loadAncestorsError: 'Could not load parents',
      },
      limits: limitTree,
      loadChildren: expect.any(Function),
      location: expect.any(Object),
      match: expect.any(Object),
      organizations: orgs,
      params: {
        is_jurisdiction: true,
        return_geometry: false,
      },
      plan: plans[0],
      propertyFilters: {
        status: 'Active',
      },
      readAPIEndpoint: 'location',
      serviceClass: expect.any(Function),
      submitCallBackFunc: expect.any(Function),
      supersetService: supersetServiceMock,
    });
    wrapper.unmount();
  });
});
