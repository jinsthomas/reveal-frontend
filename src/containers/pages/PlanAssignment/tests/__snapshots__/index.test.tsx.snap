// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanAssignment requires plan id to work: no plan id 1`] = `null`;
